src/mapgen_functions.cpp

```diff
@@ -175,16 +175,19 @@ void init_mapgen_builtin_functions() {
 
     mapgen_cfunction_map["subway_straight"]    = &mapgen_subway_straight;
     mapgen_cfunction_map["subway_curved"]      = &mapgen_subway_curved;
+    mapgen_cfunction_map["subway_end"]         = &mapgen_subway_straight;   // @todo Implement
```

Comment:
They're not really dead-ends (though they could be), they simply duplicate straight segments to eliminate the special case. I'll document this better.